lib/cwrap.c

```diff
@@ -0,0 +1,245 @@
+/*
+ * cwrap.c
+ *
+ *  Created on: Oct 17, 2016
+ *      Author: ujay
+ */
+#include <limits.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include"../include/cwrapHeader.h"
+#include <limits.h>
+#include <sys/types.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include<assert.h>
+
+
+static Map* map;
+
+Map *initializeMap(int capacity){
+	int i;
+	Map *map=(Map*)malloc(sizeof(Map));
+
+	map->opened_files=(opened_dir_struct*)malloc(capacity*sizeof(opened_dir_struct));
+	assert(map->opened_files!=NULL);
+	map->capacity=capacity;
+	map->length=0;
+	return map;
+
+}
+
+Map* getMap(){
+	if(map==0){
+		map=initializeMap(4);
+	}
+	return map;
+}
+//split file from path
+char* split_path_file(char *relative_path) {
+    const char slash='/';
+   	char *filename;
+	char *dirName;
+	filename= strrchr(relative_path, slash);
+	dirName=strndup(relative_path,strlen(relative_path)- strlen(filename));
+	return dirName;
+}
+//check the capacity of map
+int checkCapacity(){
```

Comment:
This function can probably be removed. If you do want to keep it, I'd give it a `po_map*` parameter rather than using the default map, and I'd probably also make it return the number of elements remaining rather than check whether we are exactly full.